### DISCUSSION TITLE
Disable snacks → problems

### DISCUSSION THREAD

**Discussion by benfleis** [LazyVim/LazyVim#5012] [General]
**tl;dr** -- disabling snacks via plugins conf a la `{ "folke/snacks.nvim", enabled = false },` breaks -- can it be marked/held as fixed requirement (until its seemingly WIP separation is complete @folke )?

**Slightly Longer:**
I am trying to integrate my half-hand-crafted neovim config & code with LazyVim, but there are too many moving parts to do it all at once.

My first move was to scan through the many extras, and disable everything that isn't already in use, or at least an obvious win that I can use right away -- this includes many visual niceties for now.

Among others I disabled snacks. Curiously doing so leaves neovim in a nearly unusable state, not sure what's going on the bg but it seems to be busy/slow.

First simple fix is to wrap this line from LazyVim options:
`opt.statuscolumn = [[%!v:lua.require'snacks.statuscolumn'.get()]]`

with something like `if require("snacks") then ...`

Next up I crashed against a fixed ref to `Snacks` in `lua/lazyvim/util/mini.lua`. There I stopped.

Anything useful to do/help with here? Will snacks be fully separable?

**Comment by folke** [comment]
LazyVim has a hard dependency on snacks.

**Comment by benfleis** [comment]
Makes reasonable sense -- is it trivial to disallow disabling? I may be the rare bird to come across it, but want to close out when possible :)

If I can be of help, lemme know!

**Reply by folke** [reply]
Like I said. It's a hard dependency. If you disable it, you're on your own.
